gangsta.js

```diff
@@ -17,8 +19,58 @@
         random = function (items) {
             return items[randmax(items.length)];
         },
+        swapChar = function (pwChar, lChars) {
+            var c, 
+            chosenChar = pwChar;
+            for (c = 0; c < lChars.length; c += 1) {
+                if (randmax(100) % 3 === 0) {
+                    chosenChar = lChars[c];
+                }
+            }
+            if (chosenChar === pwChar) {
+                chosenChar = (randmax(100) % 4 === 0) ? 
+                    pwChar.toUpperCase() : pwChar;
+            }
+            return chosenChar;
+        },
+        leetspeak = function (pw) {
+            var i;
+            pw = pw.split('');
+
+            for(i = 0; i < pw.length; i += 1) {
+                if (pw[i] === 'a' || pw[i] === 'A'){
```

Comment:
Instead of doing a bunch of conditionals here, I ou could do something like this:

```javascript
    for (i = 0; i< pw.length; i += 1) {
        pw[i] = swapChar(pw[i]);
    }
```